core/src/test/java/org/axonframework/eventsourcing/eventstore/EventUtilsTest.java

```diff
@@ -77,6 +77,13 @@ public void testDomainEventStream_lastSequenceNumberEqualToLastProcessedEntryAft
         assertEquals(Long.valueOf(1L), eventStream.getLastSequenceNumber());
     }
 
+    @Test(expected = NullPointerException.class)
+    public void testDomainEventStream_nullPointerExceptionOnEmptyEventStream() {
+        DomainEventStream eventStream = EventUtils.upcastAndDeserializeDomainEvents(Stream.empty(),
+            serializer, NoOpEventUpcaster.INSTANCE, false);
+        long lastSequenceNumber = eventStream.getLastSequenceNumber();
```

Comment:
I would prefer the method to return `null` (and thus declare `Long` return type instead.